extensions/amp-auto-ads/0.1/ad-network-config.js

```diff
@@ -123,3 +127,73 @@ class AdSenseNetworkConfig {
     };
   }
 }
+
+
+/**
+ * @implements {AdNetworkConfigDef}
+ */
+class DoubleclickNetworkConfig {
+  /**
+   * @param {!Element} autoAmpAdsElement
+   */
+  constructor(autoAmpAdsElement) {
+    this.autoAmpAdsElement_ = autoAmpAdsElement;
+  }
+
+  /**
+   * @param {!Window} win
+   */
+  isEnabled(win) {
+    const branch = getAdSenseAmpAutoAdsExpBranch(win);
+    return branch != AdSenseAmpAutoAdsHoldoutBranches.CONTROL;
+  }
+
+  /** @override */
+  getConfigUrl() {
+    const docInfo = Services.documentInfoForDoc(this.autoAmpAdsElement_);
+    const canonicalHostname = parseUrlDeprecated(docInfo.canonicalUrl).hostname;
+    return buildUrl('//pagead2.googlesyndication.com/getconfig/ama', {
+      'client': this.autoAmpAdsElement_.getAttribute('data-ad-legacy-client'),
+      'plah': canonicalHostname,
+      'ama_t': 'amp',
+      'url': docInfo.canonicalUrl,
+    }, 4096);
+  }
+
+  /** @override */
+  getAttributes() {
+    const attributes = dict({
+      'type': 'doubleclick',
+      'data-slot': this.autoAmpAdsElement_.getAttribute('data-slot'),
+      'json': this.autoAmpAdsElement_.getAttribute('data-json'),
```

Comment:
Should the key here be 'data-json'?

Reply:
It is 'json' in the doubleclick tag, but we are using 'data-json' for the amp-auto-ads tag which seems more correct, TBH. However, there are two options here:
1. Use data-json so there are no changes to the validator, and it probably will hold in the future.
2. Move the json value to the experiment as well (as proposed in the description I think).

I'm ok either way.

Reply:
OK cool. Just checking it was intentional. I could have been less lazy and looked at https://github.com/ampproject/amphtml/blob/master/extensions/amp-ad-network-doubleclick-impl/amp-ad-network-doubleclick-impl-internal.md, so sorry for that.